### ISSUE TITLE
Please add Qt6 support

### ISSUE THREAD

**Issue by DAP-DarkneSS** [lxqt/qtermwidget#436]
##### Steps to Reproduce (for bugs)

1. cmake(Qt6LinguistTools) & cmake(Qt6Widgets)
2. /usr/bin/cmake /home/abuild/rpmbuild/BUILD/qtermwidget-0.17.0/. '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=lib64 -DCMAKE_INSTALL_LIBEXECDIR=/usr/libexec -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG' '-DCMAKE_CXX_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG' '-DCMAKE_Fortran_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG' '-DCMAKE_EXE_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_MODULE_LINKER_FLAGS=-flto=auto -Wl,--as-needed' '-DCMAKE_SHARED_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/lib64/cmake/qtermwidget-qt5 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DPULL_TRANSLATIONS=no -DUSE_UTF8PROC=yes
[   24s] -- The C compiler identification is GNU 11.1.1
[   24s] -- The CXX compiler identification is GNU 11.1.1
[   24s] -- Detecting C compiler ABI info
[   24s] -- Detecting C compiler ABI info - done
[   24s] -- Check for working C compiler: /usr/bin/cc - skipped
[   24s] -- Detecting C compile features
[   24s] -- Detecting C compile features - done
[   24s] -- Detecting CXX compiler ABI info
[   24s] -- Detecting CXX compiler ABI info - done
[   24s] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[   24s] -- Detecting CXX compile features
[   24s] -- Detecting CXX compile features - done
[   24s] CMake Error at CMakeLists.txt:37 (find_package):
[   24s]   By not providing "FindQt5Widgets.cmake" in CMAKE_MODULE_PATH this project
[   24s]   has asked CMake to find a package configuration file provided by
[   24s]   "Qt5Widgets", but CMake did not find one.
[   24s] 
[   24s]   Could not find a package configuration file provided by "Qt5Widgets"
[   24s]   (requested version 5.12.0) with any of the following names:
[   24s] 
[   24s]     Qt5WidgetsConfig.cmake
[   24s]     qt5widgets-config.cmake
[   24s] 
[   24s]   Add the installation prefix of "Qt5Widgets" to CMAKE_PREFIX_PATH or set
[   24s]   "Qt5Widgets_DIR" to a directory containing one of the above files.  If
[   24s]   "Qt5Widgets" provides a separate development package or SDK, be sure it has
[   24s]   been installed.
[   24s] 
[   24s] 
[   24s] -- Configuring incomplete, errors occurred!

##### System Information

* Distribution & Version: openSUSE Factory (rolling)
* Kernel: 5.13.6
* Qt Version: 6.2
* lxqt-build-tools Version: 0.9.0
* Package version: 0.17.0

**Comment by yan12125** [comment]
I have a work-in-progress branch at https://github.com/lxqt/qtermwidget/tree/wip-qt6. It is far from finished, though.

See also https://github.com/lxqt/lxqt/discussions/1901

**Comment by DAP-DarkneSS** [comment]
> https://github.com/lxqt/qtermwidget/tree/wip-qt6

Qt5LinguistTools is still required by cmake.
[_log.txt](https://github.com/lxqt/qtermwidget/files/6973172/_log.txt)

**Comment by yan12125** [comment]
Yep, https://github.com/lxqt/lxqt-build-tools/blob/master/cmake/modules/LXQtTranslateTs.cmake.in is not updated for Qt 6 yet.

**Comment by yan12125** [comment]
Here comes the first major issue - KPty is not compatible with Qt 6 yet https://phabricator.kde.org/T13940. Although theoretically we can move forward without waiting for KDE devs, nobody in LXQt understands kptyprocess well as far as I know.

**Comment by DAP-DarkneSS** [comment]
It sounds OK for me to work with KDE.

**Comment by Conan-Kudo** [comment]
KPty is now ported to Qt6, and at least in Fedora 40, exists as a framework package `kf6-kpty`.

**Comment by yan12125** [comment]
Yes, we can now learn from KF6 kpty. There are WIP pull requests at https://github.com/lxqt/qtermwidget/pull/515 and https://github.com/lxqt/qterminal/pull/1067.

**Comment by yan12125** [comment]
Qt6 and PyQt6 are now supported! Although there are still some minor glitches, qterminal and qtermwidget are usable with Qt 6 in general.
